### PR TITLE
[dropbear] patch to allow g+w and o+w to the user's homedir, ~/.ssh, and ~/.ssh/authorized_keys

### LINKED ISSUE #5661: SSH via local companion not working on Ubuntu
### Bug description

First time attempting to use the local companion, something seems to be off with SSH auth.

Prior to running anything, I've had default keys set up (and working) for quite some time. The companion runs normally, connecting to the workspace and populating keys and config files:

```bash
jkaye@jkaye-linux ~/Downloads $ ls /tmp/gitpod_*
/tmp/gitpod_1a60521e-cb24-4723-b7f6-3fabd5645abc_id_rsa      /tmp/gitpod_d45ebe9d-3ff3-4eaa-a21f-7060164a2bc2_id_rsa      /tmp/gitpod_ssh_config
/tmp/gitpod_1a60521e-cb24-4723-b7f6-3fabd5645abc_id_rsa.pub  /tmp/gitpod_d45ebe9d-3ff3-4eaa-a21f-7060164a2bc2_id_rsa.pub

jkaye@jkaye-linux ~/Downloads $ cat /tmp/gitpod_ssh_config 
Host tomato-cat-gv5z3x4g
HostName 127.0.0.1
User gitpod
Port 33427
IdentityFile /tmp/gitpod_1a60521e-cb24-4723-b7f6-3fabd5645abc_id_rsa
```

I tried starting a new pod, so you see the second set of keys there.

Then, the unfortunate ssh command:

```bash
jkaye@jkaye-linux ~/Downloads $ ssh -F /tmp/gitpod_ssh_config tomato-cat-gv5z3x4g
gitpod@127.0.0.1: Permission denied (publickey).
```

### Steps to reproduce

Ubuntu 18.04.5, start the local companion and attempt to connect via SSH.

### Expected behavior

_No response_

### Example repository

_No response_

### Anything else?

I tried a few different things: directly setting the identity file, etc. I've also verified that the authorized public key uploaded to the workspace matches the public key listed in the auto-generated configuration file.

### PR DESCRIPTION
**Pull request by akosyakov**
## Description

We are don't expect other users beside root and gitpod, so it should be fine to loosen permission constaints by dropbear.

## Related Issue(s)

fix #5661

## How to test

- Start a workspace.
- In terminal: `chmod g+w /home/gitpod`
- In VS Code Web, select to open in VS Code Desktop. You should be able to connect.

## Release Notes

```release-note
NONE
```

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **akosyakov** after the PR has been reviewed.

Associated issue: *#566*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/supervisor/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/supervisor/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by jeanp413 — APPROVED** [review]
/lgtm

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: cd07b4ae20f449ea3a3ae373246eb582d99d2df2</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/5768#pullrequestreview-758986591" title="LGTM">jeanp413</a>*

Associated issue: *#5661*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/supervisor/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/supervisor/OWNERS)~~ [jeanp413]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
